opentelemetry-exporter-google-cloud/tests/test_cloud_monitoring_exporter_integration.py

```diff
@@ -13,56 +13,70 @@
 # limitations under the License.
 
 
+import logging
+from unittest.mock import MagicMock, patch
+
 import grpc
 from google.cloud.monitoring_v3 import MetricServiceClient
 from google.cloud.monitoring_v3.gapic.transports import (
     metric_service_grpc_transport,
 )
 from opentelemetry.exporter.cloud_monitoring import (
-    NANOS_PER_SECOND,
-    WRITE_INTERVAL,
     CloudMonitoringMetricsExporter,
 )
 from opentelemetry.sdk import metrics
-from opentelemetry.sdk.metrics.export import MetricRecord, MetricsExportResult
-from opentelemetry.sdk.metrics.export.aggregate import SumAggregator
+from opentelemetry.sdk.metrics.export.controller import PushController
 from opentelemetry.sdk.resources import Resource
 
 from test_common import BaseExporterIntegrationTest
 
+logger = logging.getLogger(__name__)
+
 
 class TestCloudMonitoringSpanExporter(BaseExporterIntegrationTest):
     def test_export(self):
         channel = grpc.insecure_channel(self.address)
         transport = metric_service_grpc_transport.MetricServiceGrpcTransport(
             channel=channel
         )
+        client = MagicMock(wraps=MetricServiceClient(transport=transport))
         exporter = CloudMonitoringMetricsExporter(
-            self.project_id, client=MetricServiceClient(transport=transport)
+            self.project_id, client=client
         )
 
-        meter = metrics.MeterProvider().get_meter(__name__)
+        meter_provider = metrics.MeterProvider(
+            resource=Resource.create(
+                {
+                    "cloud.account.id": "some_account_id",
+                    "cloud.provider": "gcp",
+                    "cloud.zone": "us-east1-b",
+                    "host.id": 654321,
+                    "gcp.resource_type": "gce_instance",
```

Comment:
Have we decided to keep this one? It's incompatible with semantic conventions, but might be okay as an additional attribute.

Also, IIRC you shared a link where AWS has introduced attributes / resources specific to AWS to the spec? Shall we try doing the same if we want to support GCP-specific resource attributes?

Reply:
> Have we decided to keep this one? It's incompatible with semantic conventions, but might be okay as an additional attribute.

I think it's better to use the semantic conventions if we aren't missing any functionality https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/issues/85. Do you have any preference on adding our own GCP resource attributes instead? (Here's the AWS PR for reference https://github.com/open-telemetry/opentelemetry-specification/pull/1099))